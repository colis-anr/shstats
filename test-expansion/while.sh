#!/bin/sh

x=42
while x=1 y=2; do
    y=3
    z=4
    echo $x $y $z
done
echo $x $y $z
