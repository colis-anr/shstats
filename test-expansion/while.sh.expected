[
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "42", [] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_CompoundCommand",
              [
                "CompoundCommand_WhileClause",
                [
                  "WhileClause_While_CompoundList_DoGroup",
                  [
                    "CompoundList_Term_Separator",
                    [
                      "Term_AndOr",
                      [
                        "AndOr_Pipeline",
                        [
                          "Pipeline_PipeSequence",
                          [
                            "PipeSequence_Command",
                            [
                              "Command_SimpleCommand",
                              [
                                "SimpleCommand_CmdPrefix",
                                [
                                  "CmdPrefix_CmdPrefix_AssignmentWord",
                                  [
                                    "CmdPrefix_AssignmentWord",
                                    [ [ "Name", "x" ], [ "Word", "1", [] ] ]
                                  ],
                                  [ [ "Name", "y" ], [ "Word", "2", [] ] ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ],
                    [
                      "Separator_SeparatorOp_LineBreak",
                      [ "SeparatorOp_Semicolon" ],
                      [ "LineBreak_Empty" ]
                    ]
                  ],
                  [
                    "DoGroup_Do_CompoundList_Done",
                    [
                      "CompoundList_NewLineList_Term_Separator",
                      [ "NewLineList_NewLine" ],
                      [
                        "Term_Term_Separator_AndOr",
                        [
                          "Term_Term_Separator_AndOr",
                          [
                            "Term_AndOr",
                            [
                              "AndOr_Pipeline",
                              [
                                "Pipeline_PipeSequence",
                                [
                                  "PipeSequence_Command",
                                  [
                                    "Command_SimpleCommand",
                                    [
                                      "SimpleCommand_CmdPrefix",
                                      [
                                        "CmdPrefix_AssignmentWord",
                                        [
                                          [ "Name", "y" ],
                                          [ "Word", "3", [] ]
                                        ]
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ],
                          [
                            "Separator_NewLineList",
                            [ "NewLineList_NewLine" ]
                          ],
                          [
                            "AndOr_Pipeline",
                            [
                              "Pipeline_PipeSequence",
                              [
                                "PipeSequence_Command",
                                [
                                  "Command_SimpleCommand",
                                  [
                                    "SimpleCommand_CmdPrefix",
                                    [
                                      "CmdPrefix_AssignmentWord",
                                      [
                                        [ "Name", "z" ],
                                        [ "Word", "4", [] ]
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ],
                        [
                          "Separator_NewLineList",
                          [ "NewLineList_NewLine" ]
                        ],
                        [
                          "AndOr_Pipeline",
                          [
                            "Pipeline_PipeSequence",
                            [
                              "PipeSequence_Command",
                              [
                                "Command_SimpleCommand",
                                [
                                  "SimpleCommand_CmdName_CmdSuffix",
                                  [
                                    "CmdName_Word",
                                    [
                                      "Word",
                                      "echo",
                                      [ [ "WordName", "echo" ] ]
                                    ]
                                  ],
                                  [
                                    "CmdSuffix_CmdSuffix_Word",
                                    [
                                      "CmdSuffix_CmdSuffix_Word",
                                      [
                                        "CmdSuffix_Word",
                                        [ "Word", "1", [] ]
                                      ],
                                      [ "Word", "3", [] ]
                                    ],
                                    [ "Word", "4", [] ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ],
                      [ "Separator_NewLineList", [ "NewLineList_NewLine" ] ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName_CmdSuffix",
                [
                  "CmdName_Word",
                  [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                ],
                [
                  "CmdSuffix_CmdSuffix_Word",
                  [
                    "CmdSuffix_CmdSuffix_Word",
                    [
                      "CmdSuffix_Word",
                      [ "Word", "$x", [ [ "WordLiteral", "$x" ] ] ]
                    ],
                    [ "Word", "$y", [ [ "WordLiteral", "$y" ] ] ]
                  ],
                  [ "Word", "$z", [ [ "WordLiteral", "$z" ] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]